src/Airship/Internal/Route.hs

```diff
@@ -80,14 +80,19 @@ star = Route [RestUnbound]
 newtype RoutingSpec m a = RoutingSpec { getRouter :: Writer [(Route, Resource m)] a }
     deriving (Functor, Applicative, Monad, MonadWriter [(Route, Resource m)])
 
-route :: [(Route, a)] -> [Text] -> a -> (a, (HashMap Text Text, [Text]))
-route routes pInfo resource404 = foldr' (matchRoute pInfo) (resource404, (mempty, mempty)) routes
+route :: [(Route, a)]
+      -> [Text]
+      -> Maybe (a, (HashMap Text Text, [Text]))
```

Comment:
I've created a simple type alias, `MatchedRoute`, that reduces some of the line noise, but I'm afraid it doesn't do much to increase clarity. I think doing anything more, however, goes beyond the scope of this PR and is better handled separately.